### PR TITLE
Add new board rev_0131 for Keychron Q0

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
As title, and update rev_0130.

## Description

- Added rev_0131 witch enables EC11 rotary encoder.
- Updated rev_0130.
1. Rewrote the light effect definition to keep up with other Keychron keyboards.
2. Added new folder which is named keychron into keymaps of rev_0130.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q0/rev_0130/keymaps/via/config.h [hunk 1]
```diff
@@ -14,25 +14,4 @@
...
-) \
-{ \
-    { K00, K01, K02, K03 }, \
-    { K10, K11, K12, K13 }, \
-    { K20, K21, K22, K23 }, \
-    { K30, K31, K32, XXX }, \
-    { K40, K41, K42, K43 }, \
-    { K50, XXX, K52, XXX }, \
-}
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (drashna) [inline keyboards/keychron/q0/rev_0130/keymaps/via/config.h:17]: This is the default value, and shouldn't need to be specified.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/rev_0130/keymaps/via/config.h (hunk 1 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q0/rev_0130/keymaps/via/config.h]: All right, deleted it, thanks~

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q0/rev_0131/rules.mk [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# MCU name
+MCU = STM32L432
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/keychron/q0/rev_0131/rules.mk:6]:
```suggestion
```

File: keyboards/keychron/q0/rev_0131/info.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+    "keyboard_name": "Keychron Q0 Plus",
+    "manufacturer": "Keychron",
+    "url": "https://github.com/Keychron",
+    "maintainer": "lalalademaxiya1",
+    "usb": {
+        "vid": "0x3434",
+        "pid": "0x0131",
+        "device_version": "1.0.0"
+    },
```
Comment (fauxpark) [inline keyboards/keychron/q0/rev_0131/info.json:10]:
```suggestion
    },
    "processor": "STM32L432",
    "bootloader": "stm32-dfu",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/rev_0131/info.json (hunk 3 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q0/rev_0131/info.json:10]: Thanks~

**Comment by keychron-dev** [comment]
@fauxpark hi, if you have time, please have a look at this, thanks

**Comment by keychron-dev** [comment]
@fauxpark thank you, and this one.

**Review by fauxpark — APPROVED** [review]

**Comment by keychron-dev** [comment]
@drashna can you help merge this? it has two approvals already.

**Pull request merged**